files/disable_inactive_freeipa_users.sh

```diff
@@ -69,59 +69,76 @@ else
+
+    # Do we have a valid timestamp in the format that IPA and LDAP use
+    # (YYYYMMDDHHMMSSZ)?
+    if [[ $create_timestamp =~ ^([[:digit:]]{4})([[:digit:]]{2})([[:digit:]]{2})([[:digit:]]{2})([[:digit:]]{2})([[:digit:]]{2})Z$ ]]; then
+      # 1. Reformat the timestamp to ISO 8601 format so the date
+      # command can understand it.
+      # 2. Use the date command to convert the ISO 8601 timestamp to
+      # seconds since the epoch.
+      create_timestamp=${BASH_REMATCH[1]}-${BASH_REMATCH[2]}-${BASH_REMATCH[3]}T${BASH_REMATCH[4]}:${BASH_REMATCH[4]}:${BASH_REMATCH[6]}Z
+      create_timestamp=$(date --date="$create_timestamp" +%s)
+
+      # This if skips any users that were created recently but have
+      # not yet logged in.  We don't want to disable their access yet.
+      if ((create_timestamp < disable_deadline)); then
+        # Extract the user's last authentication timestamp from the
+        # LDAP output.  Note that this timestamp may not exist if the
+        # user has never logged in.
+        #
+        # --quiet means no printing unless the p command is used
+        last_authentication_timestamp=$(sed --quiet 's/^krbLastSuccessfulAuth:\s//p' <<< "$ldapsearch_output")
+
+        # Do we have a valid timestamp in the format that IPA and LDAP
+        # use (YYYYMMDDHHMMSSZ)?
+        if [[ $last_authentication_timestamp =~ ^([[:digit:]]{4})([[:digit:]]{2})([[:digit:]]{2})([[:digit:]]{2})([[:digit:]]{2})([[:digit:]]{2})Z$ ]]; then
+          # 1. Reformat the timestamp to ISO 8601 format so the date
+          # command can understand it.
+          # 2. Use the date command to convert the ISO 8601 timestamp
+          # to seconds since the epoch.
+          last_authentication_timestamp=${BASH_REMATCH[1]}-${BASH_REMATCH[2]}-${BASH_REMATCH[3]}T${BASH_REMATCH[4]}:${BASH_REMATCH[4]}:${BASH_REMATCH[6]}Z
+          last_authentication_timestamp=$(date --date="$last_authentication_timestamp" +%s)
+        else
+          # The timestamp may not exist, but it should never exist but
+          # not match the regex above.
+          if [[ -n $last_authentication_timestamp ]]; then
+            echo User "$user" has an invalid last authentication timestamp.
+          fi
         fi
-      fi
-    done
 
-    # Now that we have analyzed the last authentication timestamps,
-    # disable users as necessary.
-    #
-    # TODO: Note also that after disable_time has elapsed with the
-    # "KDC:Disable Last Success" feature disabled we can start
-    # disabling users with all authentication timestamps invalid (N/A)
-    # and simplify this logic.  See #74 for more details.
-    if $disable_user; then
-      if ! $all_timestamps_invalid; then
-        ipa user-disable "$user"
-        echo User "$user" disabled due to inactivity.
+        # If the timestamp predates the deadline, or if the user has
+        # never authenticated at all, then the user is inactive should
+        # be disabled.
```

Comment:
Add missing word in comment:
```suggestion
        # never authenticated at all, then the user is inactive and
        # should be disabled.
```